File: Aether.Core/Models/DashboardView.cs
namespace Aether.Core.Models
{
    public sealed class DashboardView
    {
        public GlobalLocation? Location { get; set; }

        public required AirQualityReading AirQualityReading { get; set; }

        public required ChangePercentages ChangePercentages { get; set; }
    }

    public class ChangePercentages
    {
        public int Aqi { get; set; }

        public double CarbonMonoxide { get; set; }
        public double SulfurDioxide { get; set; }
        public double NitrogenDioxide { get; set; }
        public double NitrogenOxide { get; set; }
        public double Ozone { get; set; }
        public double ParticulateMatter10 { get; set; }
        public double ParticulateMatter2_5 { get; set; }
        public double Ammonia { get; set; }
    }
}

File: Aether/client-app/src/app/models/dashboard-view.ts
import { GlobalLocation } from "./global-location";

export interface DashboardView {
  location: GlobalLocation;
  airQualityReading: AirQualityReading;
  changePercentages: ChangePercentages;
}

export interface AirQualityReading {
  id: number;
  location: AirQualityLocation;
  index: number;
  aqi: number;
  sulfurDioxide: Pollutant;
  nitrogenOxide: Pollutant;
  nitrogenDioxide: Pollutant;
  particulateMatter10: Pollutant;
  particulateMatter2_5: Pollutant;
  ozone: Pollutant;
  carbonMonoxide: Pollutant;
  ammonia: Pollutant;
  lastUpdated: string;
}

export interface AirQualityLocation {
  name: string;
  latitude: number;
  longitude: number;
}

export interface Pollutant {
  name: string;
  index: number;
  concentration: number;
  max: number;
}

interface ChangePercentages {
  aqi: number;
  carbonMonoxide: number;
  sulfurDioxide: number;
  nitrogenDioxide: number;
  nitrogenOxide: number;
  ozone: number;
  particulateMatter10: number;
  particulateMatter2_5: number;
  ammonia: number;
}
